build.sh

```diff
@@ -178,6 +184,49 @@ setup_build_environment() {
   BINDIR="${BINROOT}/${FULL_VARIANT}/${OUTDIR}"
 }
 
+#-----------------------------------------------------------------------------
+# Function: prepare_coverage_capture
+# Run lcov preparations before testing for coverage
+#-----------------------------------------------------------------------------
+prepare_coverage_capture() {
+  [[ -n $GITHUB_ACTIONS ]] && echo "::group::Code Coverage Preparation"
+  lcov --zerocounters      --directory $BINROOT --base-directory $ROOT
+  lcov --capture --initial --directory $BINROOT --base-directory $ROOT -o $BINROOT/base.info
+  [[ -n $GITHUB_ACTIONS ]] && echo "::endgroup::"
+}
+
+#-----------------------------------------------------------------------------
+# Function: capture_coverage
+# Capture coverage collected since `prepare_coverage_capture` was invoked
+#-----------------------------------------------------------------------------
+capture_coverage() {
+  NAME=${1:-cov} # Get output name. Defaults to `cov.info`
+
+  [[ -n $GITHUB_ACTIONS ]] && echo "::group::Code Coverage Capture ($NAME)"
+
+  # Capture coverage collected while running tests previously
+  lcov --capture --directory $BINROOT --base-directory $ROOT -o $BINROOT/test.info
+
+  # Accumulate results with the baseline captured before the test
+  lcov --add-tracefile $BINROOT/base.info --add-tracefile $BINROOT/test.info -o $BINROOT/full.info
+
+  # Extract only the coverage of the project source files
+  lcov --output-file $BINROOT/$NAME.info --extract $BINROOT/full.info \
+    "$ROOT/src/*" \
+    "$ROOT/deps/triemap/*" \
+    "$ROOT/deps/thpool/*" \
+
+  # TODO: remove `tests`?
+  # # Remove coverage for directories we don't want
+  # lcov -o $BINROOT/tmp.info --remove $BINROOT/full.info \
+  #   "*/tests/*" \
```

Comment:
What is that? Why should we care about coverage of tests?

Reply:
We don't, but we are also not collecting such info today (we don't have `tests` sub-directory under `src`, except for rust code which we collect "automatically" thanks to `cargo`, and not by this manual implementation.
On 2.10 and previous builds, we have such directories (for the coordinator for example), so we will need this step

Reply:
I uncommented the step to be aligned on all branches, and also ignore future `tests` subdirectories